src/Lean/Message.lean

```diff
@@ -69,13 +60,13 @@ partial def formatAux : NamingContext → Option MessageDataContext → MessageD
 | _,    _,         ofFormat fmt             => pure fmt
 | _,    _,         ofLevel u                => pure $ fmt u
 | _,    _,         ofName n                 => pure $ fmt n
-| _,    some ctx,  ofSyntax s               => pure $ s.formatStx (getSyntaxMaxDepth ctx.opts)
+| nCtx, some ctx,  ofSyntax s               => ppTerm (mkPPContext nCtx ctx) s  -- HACK: might not be a term
```

Comment:
@leodemoura The pretty printer has separate functions `ppTerm/ppCommand/...`, but when pretty printing `Message.ofSyntax`, we do not know the syntax category we should call the pretty printer for. We could store it in the Message, but I'd rather simplify the pretty printer so there is a single `ppSyntax`. The only reason we need to know the category to be pretty printed is that the parenthesizer needs to know which parentheses to apply. I can see two solutions here:
* We make sure we use the same parenthesis node everywhere (i.e. terms and levels, currently). This assumes that we will really only ever need the category for parenthesization. And it's not clear to me how we would change the term parenthesis parser.
* We specify the category at `trailingNode`, which is the only point where the pretty printer [has to descend](https://github.com/kha/lean4/blob/synsan/src/Lean/PrettyPrinter/Parenthesizer.lean#L375-L379) into a category parser without the grammar telling it which one it is. We could use a syntax like `tparser! stx "?"` and `tparser! stx:2 " <|> " >> syntaxParser 1` for the macro. I like this approach more because it is more general and the new `tparser!` syntax is actually closer to our `syntax` syntax.

Reply:
When we are updating the trailing parser table, we know the parser category. We can wrap this information there before we perform the actual update.
```lean
def addTrailingParser (categories : ParserCategories) (catName : Name) (p : TrailingParser) (prio : Nat) : Except String ParserCategories := ...
def addBuiltinTrailingParser (catName : Name) (declName : Name) (p : TrailingParser) (prio : Nat) : IO Unit := ...
```

Reply:
In the functions above, we update `p` with the `catName` information. I think this should work, right?

Reply:
Perhaps it is too hackish :)

Reply:
That being said, it feels weird to provide this information manually, when it is there on the attribute.
Examples:
```lean
@[builtinTermParser] def add   := tparser! infixL " + "  65
@[builtinTermParser] def sub   := tparser! infixL " - "  65
@[builtinSyntaxParser] def orelse        := tparser!:2 " <|> " >> syntaxParser 1
```

Reply:
BTW, we have been planning to use the `syntax` notation for builtin parsers after we switch to the new frontend. Perhaps, we can ignore this issue for now.

Reply:
Yes, that's true. Do you see a good way to get at that information?

Reply:
Yeah, let's ignore it for now. For the actual formatting, it shouldn't matter which parentheses we use, only for re-elaboration, which we're not doing yet.